backend/src/main/java/com/speakfluid/backend/entities/ConfidenceScoreCalculator.java

```diff
@@ -21,20 +23,29 @@ public class ConfidenceScoreCalculator implements Scorable{
      * @param step the type of the talk step whose methods are called.
      */
 
-    public void passDialogueToTalkStep(Dialogue dialogue, TalkStep step) {
+    public void passDialogueToTalkStep(Dialogue<?> dialogue, TalkStep step) {
         step.runAnalysis(dialogue);
         stepTotalScore = step.getMaxScore();
         stepScoreAccumulator = step.getScoreAccumulator();
         step.setZeroScoreAccumulator();
     }
 
     /**
-     * Calculates the confidence score of the talk step.
+     * Calculates the confidence score of the talk step. If it exceeds 100%, then
+     * 95% will be returned as a capping mechanism.
      * @return returns the confidence score of the talk step.
      */
     @Override
     public double calculateConfidenceScore() {
-        return (stepScoreAccumulator / stepTotalScore) * 100;
+
+        if(((stepScoreAccumulator / stepTotalScore) * 100) > 100){
+            return 95.0;
+
+        }
+        else{
+            double confidenceScore = (stepScoreAccumulator / stepTotalScore) * 100;
+            return Math.round(confidenceScore * 100.0) / 100.0;
```

Comment:
why are we dividing by 100 here? doesn't Math.round(confidenceScore * 100.0) give us the confidence score in percentage already or am i doing the math wrong